homeassistant/components/arcam_fmj/config_flow.py

```diff
@@ -16,12 +85,86 @@ class ArcamFmjFlowHandler(config_entries.ConfigFlow):
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
 
+    async def _async_set_unique_id_and_update(self, host, port, uuid):
+        config = {
+            CONF_HOST: host,
+            CONF_PORT: port,
+        }
+        entry = await self.async_set_unique_id(uuid)
+        if entry:
+            if config.items() <= entry.data.items():
+                raise AbortFlow("already_setup")
+            entry.data = config
```

Comment:
don't update entry directly. Pass new data to `hass.config_entries.async_update_entry`